### ISSUE TITLE
How to enable eslint intellisense / suggestions for the rules property in .eslintrc.js

### ISSUE THREAD

**Issue by numanaral** [microsoft/vscode-eslint#1122]
I wanted to see the rules and their props via intellisense and modify them without having to go to the eslint website. I came across #124 which provides the following solution but it's only displaying the first level suggestions and not the inner ones:

### Referenced Solution:
> Sorry, I didn't see the original question. Today I think your best bet would be to install `@types/eslint` and use a JSDoc annotation to get completions.
```js
/**
 * @type {import("eslint").Linter.Config}
 */
const config = {
  // ...
}
module.exports = config;
```

_Originally posted by @DanielRosenwasser in https://github.com/microsoft/vscode-eslint/issues/124#issuecomment-609360052_

### Result:
![image](https://user-images.githubusercontent.com/25233323/99778550-11cb4e00-2ad1-11eb-908a-ea4c554be4dd.png)
![image](https://user-images.githubusercontent.com/25233323/99779205-f6ad0e00-2ad1-11eb-97d1-86e57fc931e0.png)

After playing around with the jsdoc `@typedef` a bit, I managed to get the intellisense to work for the "rules" as well with the following `@typedef`:

### Modified Solution:
```js
/**
 * @typedef {Object} ESLintRules
 * @property {import('eslint/rules').ESLintRules} rules
 * @typedef {import('eslint').Linter.Config&ESLintRules} ESLintConfig
 */

/**
 * @type {ESLintConfig}
 */
module.exports = {
    // ...your eslint config
}
```

### Result:
![image](https://user-images.githubusercontent.com/25233323/99778584-1b54b600-2ad1-11eb-8ac1-ec8a167585e2.png)
![image](https://user-images.githubusercontent.com/25233323/99778914-8dc59600-2ad1-11eb-8783-c93bb5a0b2da.png)

Posting here with the hope that it can prove helpful for someone!

**Comment by dbaeumer** [comment]
@numanaral thanks. 

However there is nothing I can do in the ESLint extension to make this work out of the box. I will therefore close the issue, but others can still access it.

**Comment by numanaral** [comment]
No problem! 

Yes, that's understandable. This is just to have an easy way out to enable intellisense :)

**Comment by betafcc** [comment]
Thanks @numanaral , I was looking for a solution like that

FYI you can shorten it:

```js
/**
 * @type {import('eslint').Linter.Config<import('eslint/rules').ESLintRules>}
 */
module.exports = {
  rules: {
    
  }
}
```

<img width="858" alt="image" src="https://user-images.githubusercontent.com/27316303/162268847-c2f608aa-a389-4d1d-85ec-abc42d28c1f1.png">

**Comment by phil294** [comment]
The above solution(s) work pretty well for rules that are part of the ones provided by ESLint itself. Plugin rules are excluded, and also this type doesn't prevent you from writing invalid rule names. In the rare case of a plugin which exposes its rules as `.d.ts`, they could be added as union to the generic argument. Tried that with https://github.com/typescript-eslint/typescript-eslint but I failed to come up with something that worked consistently.

But anyway, having some of this stuff documented would be great.
